### PR TITLE
Docs: Convert current pbench MarkDown content to readthedocs and reStructuredText format

### PR DESCRIPTION
**Pull request by haoweiqiu**
This PR converts current pbench MarkDown content to readthedocs and reStructuredText format.
The ReadtheDocs link is: https://pbench-docs.readthedocs.io/

### CODE REVIEW

**Comment by portante** [comment]
Thanks @haoweiqiu!  I'll review soon.

**Comment by portante** [comment]
Can you also consider the `gh-pages` branch content?  Though we are looking to merge https://github.com/distributed-system-analysis/pbench/pull/1586 soon.

**Comment by haoweiqiu** [comment]
@portante Thank you very much for the comment! I'll look into #1586 and modify Read the Docs.

**Review by dbutenhof — COMMENTED** [review]
Just skimmed this; comments and questions on content and formatting.

Maybe I'm unclear on the context of this review, but I see missing "pbench-" prefixes on tools, jumbled sets of command lines without line breaks, and odd sample output including Pbench debug statements. Just a few examples are marked in this review. I'll read more later.

**Review by portante — CHANGES_REQUESTED** [review]
Good start.

This adds a new directory, `docs`.  What are you thinking we should do with the existing `doc` directory?  I was hoping we could make these changes in the `doc` directory, just moving all of the current "mark down" files we have (`.md`) to reStructuredText that is supported by GitHub.

Does GitHub's rendering of the new documents work as expected?

File: docs/conf.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+#
+# import os
+# import sys
+# sys.path.insert(0, os.path.abspath('.'))
+
+
+# -- Project information -----------------------------------------------------
+
+project = 'pbench-docs'
+copyright = '2020, Haowei'
```
Comment (ndokos) [inline docs/conf.py:23]: Shouldn't the copyright be assigned to a RH entity?

File: docs/pbench-agent.rst [hunk 2]
```diff
@@ -0,0 +1,1348 @@
...
+
+   pbench_fio
+
+When the benchmark finishes, the tools will be stopped as well. The
+results can be collected and shipped to the standard storage
+location [3]_ with
+
+::
+
+   move-results
```
Comment (ndokos) [inline docs/pbench-agent.rst:163]: pbench- prefix missing

File: docs/pbench-agent.rst [hunk 3]
```diff
@@ -0,0 +1,1348 @@
...
+
+or
+
+::
+
+   copy-results
```
Comment (ndokos) [inline docs/pbench-agent.rst:169]: pbench- prefix missing

File: docs/pbench-agent.rst [hunk 4]
```diff
@@ -0,0 +1,1348 @@
...
+benchmark script, you may be able to use the ``user-benchmark`` script:
+it takes a command as argument, starts the collection tools, invokes the
+command, stops the collection tools and postprocesses the results. So
+the workflow becomes:
+
+::
+
+   register-tool-set
+   user-benchmark --config=foo -- myscript.sh
+   move-results
```
Comment (ndokos) [inline docs/pbench-agent.rst:184]: pbench- prefix missing on last three lines

File: docs/pbench-agent.rst [hunk 5]
```diff
@@ -0,0 +1,1348 @@
...
+
+   pbench_run=/var/lib/pbench
+   pbench_log=/var/lib/pbench/pbench.log
+   date=`date --utc "+%Y.%m.%dT%H.%M.%S"`
+   hostname=`hostname -s`
+   results_repo=pbench@pbench.example.com
+   results_repo_dir=/pbench/public_html/incoming
+   ssh_opts='-o StrictHostKeyChecking=no'
+
+These are now specified in the config file
```
Comment (ndokos) [inline docs/pbench-agent.rst:220]: This whole section does not sound right: `base` does set a bunch of variables but most of these are not specified in the config file. In particular, `date`, `hostname`, `results_repo` and `results_repo_dir` are not specified in the config file. The last two I can't find anywhere, either in the config file or in `base`.

File: docs/pbench-agent.rst [hunk 6]
```diff
@@ -0,0 +1,1348 @@
...
+``/opt/pbench-agent/config/pbench.conf``.
```
Comment (ndokos) [inline docs/pbench-agent.rst:221]: The config file is `/opt/pbench-agent/config/pbench-agent.cfg`.

File: docs/pbench-agent.rst [hunk 7]
```diff
@@ -0,0 +1,1348 @@
...
+
+Available tools
+===============
+
+The configured default set of tools (what you would get by running
+``register-tool-set``) is:
```
Comment (ndokos) [inline docs/pbench-agent.rst:227]: pbench-

**Review by ndokos — CHANGES_REQUESTED** [review]
It seems that this is based on a very old set of docs that does not follow current conventions. It needs a fair amount of alterations (the lack of the `pbench-` prefix on most commands is the most obvious omission).

**Comment by ndokos** [comment]
What is the process of publishing these documents on `readTheDocs`?

**Comment by pravins** [comment]
This has been fixed with https://github.com/distributed-system-analysis/pbench/pull/3428
Feel free to reopen if something is missing.

**Pull request closed without merging**